modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaCaskServerCodegen.java

```diff
@@ -777,7 +781,7 @@ private static String defaultValueNonOption(IJsonSchemaValidationProperties p, S
         if (p.getIsMap()) {
             return "Map.empty";
         }
-        if (p.getIsNumber()) {
+        if (p.getIsNumber() || p.getIsFloat() || p.getIsDecimal() || p.getIsDouble() || p.getIsInteger()  || p.getIsLong() || p.getIsUnboundedInteger()) {
```

Comment:
what about using `isNumeric` instead?

Reply:
That's a fair shout. Originally the signature of 'defaultValue' took an 'IJsonSchemaValidationProperties' (which doesn't have 'isNumeric'). It turns out I can narrow the method signature to make 'isNumeric' available.

Cheers!